test/spec/modules/ttdBidAdapter_spec.js

```diff
@@ -0,0 +1,1090 @@
+      expect(requestBody.site.page).to.equal('https://www.example.com/test');
+    });
+
+    it('sets the banner pos correctly if sent', function () {
+      let clonedBannerRequests = deepClone(baseBannerBidRequests);
+      clonedBannerRequests[0].mediaTypes.banner.pos = 1;
+
+      const requestBody = testBuildRequests(clonedBannerRequests, baseBidderRequest).data;
+      expect(requestBody.imp[0].banner.pos).to.equal(1);
+    });
+
+    it('sets the banner expansion direction correctly if sent', function () {
+      let clonedBannerRequests = deepClone(baseBannerBidRequests);
+      const expdir = [1, 3]
+      clonedBannerRequests[0].params.banner = {
+        expdir: expdir
+      };
+
+      const requestBody = testBuildRequests(clonedBannerRequests, baseBidderRequest).data;
+      expect(requestBody.imp[0].banner.expdir).to.equal(expdir);
+    });
+
+    it('sets categories correctly if sent', function () {
+      let clonedBannerRequests = deepClone(baseBannerBidRequests);
+      const categories = ['IAB8-5', 'IAB14-3'];
+      clonedBannerRequests[0].params.categories = categories;
+
+      const requestBody = testBuildRequests(clonedBannerRequests, baseBidderRequest).data;
+      expect(requestBody.site.pagecat).to.equal(categories);
+    });
+
+    it('sets keywords properly if sent', function () {
+      let clonedBannerRequests = deepClone(baseBannerBidRequests);
+      const keywords = ['highViewability', 'clothing', 'holiday shopping'];
+      clonedBannerRequests[0].params.keywords = keywords;
+
+      const requestBody = testBuildRequests(clonedBannerRequests, baseBidderRequest).data;
+      expect(requestBody.ext.ttdprebid.keywords).to.deep.equal(keywords);
+    });
+
+    it('sets ext properly', function () {
+      let clonedBannerRequests = deepClone(baseBannerBidRequests);
+
+      const requestBody = testBuildRequests(clonedBannerRequests, baseBidderRequest).data;
+      expect(requestBody.ext.ttdprebid.pbjs).to.equal('$prebid.version$');
+    });
+
+    it('sets secure correctly if sent', function () {
+      let clonedBannerRequests = deepClone(baseBannerBidRequests);
+      clonedBannerRequests[0].params.secure = 1;
+
+      const requestBody = testBuildRequests(clonedBannerRequests, baseBidderRequest).data;
+      expect(requestBody.imp[0].secure).to.equal(1);
+    });
+
+    it('adds gdpr consent info to the request', function () {
+      let consentString = 'BON3G4EON3G4EAAABAENAA____ABl____A';
+      let clonedBidderRequest = deepClone(baseBidderRequest);
+      clonedBidderRequest.gdprConsent = {
+        consentString: consentString,
+        gdprApplies: true
+      };
+
+      const requestBody = testBuildRequests(baseBannerBidRequests, clonedBidderRequest).data;
+      expect(requestBody.user.ext.consent).to.equal(consentString);
+      expect(requestBody.regs.ext.gdpr).to.equal(1);
+    });
+
+    it('adds usp consent info to the request', function () {
```

Comment:
might as well add a test for coppa too right?

Reply:
added